src/Diagnostics.RuntimeHost/Services/GithubClient/GithubClient.cs

```diff
@@ -87,13 +94,33 @@ public async Task DownloadFile(string fileUrl, string destinationPath)
                 }
 
                 using (Stream srcStream = await httpResponse.Content.ReadAsStreamAsync(),
-                        destStream = new FileStream(destinationPath, FileMode.Create))
+                        destStream = new FileStream(destinationPath, System.IO.FileMode.Create))
                 {
                     await srcStream.CopyToAsync(destStream);
                 }
             }
         }
 
+        public async Task CreateOrUpdateFile(string filePath, string content, string commitMessage, bool convertContentToBase64 = true)
+        {
+            try
+            {
+                // try to get the file (and with the file the last commit sha)
+                var existingFile = await _octokitClient.Repository.Content.GetAllContentsByRef(_userName, _repoName, filePath, _branch);
+
+                // update the existing file
+                var updateChangeSet = await _octokitClient.Repository.Content.UpdateFile(_userName, _repoName, filePath,
+                   new UpdateFileRequest(commitMessage, content, existingFile.First().Sha, _branch, convertContentToBase64));
```

Comment:
Shekhar what will happen if there's  file existing but no actual changes happened? Will it be update too?

Reply:
Ideally, that call shouldn't come through from applens. But lets say such call comes through. Then there would be a call to update the file with same content.
However, the file SHA will not be modified (since content is same) and we will not sync that file from github (as github will respond with no changes detected)